consensus/scripts/nbconverted/build-consensus-signatures.py

```diff
@@ -179,31 +196,108 @@ def consensus_apply(df, operation, cp_features, replicate_cols):
+        )
+
+        # How many features in feature selected profile?
+        print(
+            f"There are {consensus_profiles[operation]['feat_select'].shape[1]} features in {operation} consensus profiles for {norm_strat} normalization"
         )
 
     all_consensus_dfs[norm_strat] = consensus_profiles
 
 
-# ## Merge and Output Consensus Signatures
+# ## Merge and Output Consensus Signatures, with and without feature selection
 
 # In[9]:
 
 
+float_format = "%5g"
+compression = "gzip"
+
 for norm_strat in file_bases:
     file_suffix = file_bases[norm_strat]["output_file_suffix"]
     for operation in operations:
+
+        # No feature selection
         consensus_file = f"{batch}_consensus_{operation}{file_suffix}"
         consensus_file = pathlib.Path(batch, consensus_file)
 
-        consensus_df = all_consensus_dfs[norm_strat][operation]
+        consensus_df = all_consensus_dfs[norm_strat][operation]["no_feat_select"]
 
         print(
-            f"Now Writing: Consensus Operation: {operation}; Norm Strategy: {norm_strat}\nFile: {consensus_file}"
+            f"Now Writing: Feature selection: No; Consensus Operation: {operation}; Norm Strategy: {norm_strat}\nFile: {consensus_file}"
         )
         print(consensus_df.shape)
 
         consensus_df.to_csv(
-            consensus_file, sep=",", compression="gzip", float_format="%5g", index=False
+            consensus_file,
+            sep=",",
+            compression=compression,
+            float_format=float_format,
+            index=False,
         )
 
+        # With feature selection
+        consensus_feat_df = all_consensus_dfs[norm_strat][operation]["feat_select"]
+
+        consensus_feat_file = (
+            f"{batch}_consensus_{operation}_feature_select{file_suffix}"
+        )
+        consensus_feat_file = pathlib.Path(batch, consensus_feat_file)
+
+        print(
+            f"Now Writing: Feature selection: Yes; Consensus Operation: {operation}; Norm Strategy: {norm_strat}\nFile: {consensus_feat_file}"
+        )
+        print(consensus_feat_df.shape)
+
+        consensus_feat_df.to_csv(
+            consensus_feat_file,
+            sep=",",
+            compression=compression,
+            float_format=float_format,
+            index=False,
+        )
+
+
+# ## Save whole plate MODZ consensus signature as GCT
+# 
+# Whole-plate-normalized + MODZ aggregated consensus profiles will be made available on clue.io/morphology as a GCT file.
+
+# In[11]:
+
+
+import pycytominer.write_gct
+
+operation = "modz"
+norm_strat = "whole_plate"
+file_suffix = ".gct"
+consensus_file = f"{batch}_consensus_{operation}{file_suffix}"
+consensus_file = pathlib.Path(batch, consensus_file)
+
+consensus_df = all_consensus_dfs[norm_strat][operation]["no_feat_select"]
+
+print(
+    f"Now Writing: Consensus Operation: {operation}; Norm Strategy: {norm_strat}\nFile: {consensus_file}"
+)
+print(consensus_df.shape)
+
+pycytominer.write_gct(consensus_df, consensus_file)
```

Comment:
is it intentional that you only create one gct file instead of for every consensus output file?

Reply:
Yep. The only reason to create a GCT file was to be able to load it directly in Morpheus. There's not much more of an upside to it beyond that.

This is a **great question** because I'm now not convinced we should include GCT at all!

1. Morpheus already accepts CSV files, which we already make available.
2. The GCT format does not present an advantage over CSV, for our purposes
3. `pycytominer.write_gct` is available if one does want to use convert to GCT, but we needn't make it available in this repo
4. Adding corresponding GCT files will increase size of this folder by 252M, more if we don't compress

In a dramatic turn of events, I propose we drop GCT files! I've modified the PR to do so; LMK if you disagree.

Reply:
If our future selves wonder why we considered GCT in the first place, this historical context might help: https://github.com/broadinstitute/lincs-cell-painting/issues/46#issuecomment-635602422

Reply:
awesome!